### PR TITLE
[FLINK-29982] move the existing Cassandra connector to external repository

### PR DESCRIPTION
**Pull request by echauchot**
Reopening of [the previous PR](https://github.com/apache/flink-connector-cassandra/pull/1) that was automatically closed by github.

R: @zentol

### CODE REVIEW

File: flink-connector-cassandra/pom.xml [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+		<junit4.version>4.13.2</junit4.version>
+		<junit5.version>5.8.1</junit5.version>
+		<assertj.version>3.21.0</assertj.version>
+		<archunit.version>0.22.0</archunit.version>
+		<testcontainers.version>1.17.2</testcontainers.version>
+		<mockito.version>2.21.0</mockito.version>
+		<slf4j.version>1.7.36</slf4j.version>
+		<log4j.version>2.17.2</log4j.version>
+		<japicmp.skip>false</japicmp.skip>
+		<japicmp.referenceVersion>1.15.0</japicmp.referenceVersion>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:51]: should be 1.16.0

File: flink-connector-cassandra/pom.xml [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+		<scala.binary.version>2.12</scala.binary.version>
+		<byte-buddy.version>1.12.19</byte-buddy.version>
+		<findbugs.version>1.3.9</findbugs.version>
+
+
+		<!--driver 3.x works with 3.x and 4.x versions of Cassandra but driver 4.x is a complete refactoring with different API-->
+		<driver.version>3.11.2</driver.version>
+		<guava.version>19.0</guava.version>
+		<scala-library.version>2.12.7</scala-library.version>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:60]: lets move this closer to scala.binary.version

File: flink-connector-cassandra/pom.xml [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-connector-cassandra-parent</artifactId>
+		<version>3.0-SNAPSHOT</version>
+	</parent>
+
+	<groupId>org.apache.flink</groupId>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:32]:
```suggestion
```
redundant

File: flink-connector-cassandra/pom.xml [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+	<artifactId>flink-connector-cassandra_${scala.binary.version}</artifactId>
+	<version>3.0-SNAPSHOT</version>
+	<name>Flink : Connectors : Cassandra</name>
+	<packaging>jar</packaging>
+
+	<!-- Allow users to pass custom connector versions -->
+	<properties>
+		<flink.version>1.16.0</flink.version>
+		<flink.shaded.version>15.0</flink.shaded.version>
+		<junit4.version>4.13.2</junit4.version>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:42]: we shouldn't need junit4

File: flink-connector-cassandra/pom.xml [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+			<scope>provided</scope>
+		</dependency>
+
+		<!-- Test dependencies -->
+
+		<dependency>
+			<groupId>org.junit.vintage</groupId>
+			<artifactId>junit-vintage-engine</artifactId>
+			<version>${junit5.version}</version>
+			<scope>test</scope>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:92]: The dependency section is a bit of a mess. Let's organize things by scope a bit.

File: flink-connector-cassandra/pom.xml [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-architecture-tests-test</artifactId>
+			<version>${flink.version}</version>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-architecture-tests-production</artifactId>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:212]: Missing production tests class (this was handled centrally in Flink before; now every connectors must define it themselves)

File: flink-connector-cassandra/pom.xml [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+			<exclusions>
+				<exclusion>
+					<groupId>log4j</groupId>
+					<artifactId>log4j</artifactId>
+				</exclusion>
+				<exclusion>
+					<groupId>org.slf4j</groupId>
+					<artifactId>slf4j-log4j12</artifactId>
+				</exclusion>
+			</exclusions>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:184]: should no longer be required (since 1.16.0)

File: flink-connector-cassandra/pom.xml [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+		<flink.shaded.version>15.0</flink.shaded.version>
+		<junit4.version>4.13.2</junit4.version>
+		<junit5.version>5.8.1</junit5.version>
+		<assertj.version>3.21.0</assertj.version>
+		<archunit.version>0.22.0</archunit.version>
+		<testcontainers.version>1.17.2</testcontainers.version>
+		<mockito.version>2.21.0</mockito.version>
+		<slf4j.version>1.7.36</slf4j.version>
+		<log4j.version>2.17.2</log4j.version>
+		<japicmp.skip>false</japicmp.skip>
```
Comment (zentol) [inline flink-connector-cassandra/pom.xml:50]: don't think we'll need this anymore

**Review by zentol — CHANGES_REQUESTED** [review]
CI is failing due to dependency convergence; we'll need to add a few entries to the dependency management section like we had to in other connectors.

**Comment by echauchot** [comment]
> CI is failing due to dependency convergence; we'll need to add a few entries to the dependency management section like we had to in other connectors.

Yes I'm fixing them right now.

File: flink-connector-cassandra/pom.xml (hunk 1 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:51]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 2 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:60]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 3 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:32]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 4 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:42]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 5 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:92]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 7 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:184]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 8 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:50]: :+1:

File: flink-connector-cassandra/pom.xml (hunk 2 above)
Comment (echauchot) [reply flink-connector-cassandra/pom.xml:60]: ah ok, I though you were talking about minor versions :smile:

**Comment by echauchot** [comment]
@zentol I finished addressing your comments and the CI is green. PTAL

**Comment by zentol** [comment]
Couldn't sleep so I just started fixing things myself. Squashed all your changes into a single commit.

File: NOTICE [hunk 9]
```diff
@@ -13,4 +13,10 @@ DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING F
 WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE
 USE OR PERFORMANCE OF THIS SOFTWARE.
 
+This project bundles the following dependencies under the Apache Software License 2.0. (http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+- com.datastax.cassandra:cassandra-driver-core:shaded:3.11.2
+- com.datastax.cassandra:cassandra-driver-mapping:3.11.2
+- com.google.guava:guava:19.0
+
```
[the remark this answers is not in the document]
Comment (echauchot) [reply NOTICE]: ah ok, thx

**Comment by echauchot** [comment]
> Couldn't sleep so I just started fixing things myself. Squashed all your changes into a single commit.

Thanks Chesnay for the review and the night merge :smile: I can now more my Cassandra source PR to this repo.

**Pull request merged**